### PR TITLE
feat: Collapse support cssVar

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #45618 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Collapse support cssVar       |
| 🇨🇳 Chinese |    折叠面板支持 css 变量       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 804ab6b</samp>

This pull request refactors the Collapse component to use CSS variables for styling, improving the code quality and maintainability. It introduces a new `useCSSVar` hook in `components/collapse/style/cssVar.ts` that registers and applies the CSS variables dynamically. It also simplifies the style code in `components/collapse/style/index.ts` by using utility functions and properties from the theme module.

### <samp>🤖 Generated by Copilot at 804ab6b</samp>

*  Import and use `useCSSVar` hook to register and apply CSS variables for Collapse component ([link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9R19), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9L79-R81), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-3541251bb31318316b91e1f6710b646bd92edaff4318ca790c6401bea6962fc9L156-R158), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-e0c4da0d6c95ed6c3adf10a4ba5bbc47f8d919186809e5f4e01ee3bf8d2b1a82R1-R4))
*  Replace hard-coded values with `unit` function to ensure consistent units for CSS variables and style properties ([link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L2-R6), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L75-R78), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L87-R92), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L310-R327))
*  Replace arithmetic expressions with `token.calc` method to simplify margin calculations ([link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L173-R178), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L191-R198))
*  Replace repeated calculations with `fontHeight` property to simplify height settings ([link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L112-R117), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L191-R198))
*  Use `prepareComponentToken` function to generate default token values for Collapse component ([link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L310-R327), [link](https://github.com/ant-design/ant-design/pull/45862/files?diff=unified&w=0#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0L327-R339))

### CODE REVIEW

File: components/collapse/style/index.ts [hunk 1]
```diff
@@ -62,17 +64,18 @@ export const genBaseStyle: GenerateStyle<CollapseToken> = (token) => {
...
+  const borderBase = `${unit(lineWidth)} ${lineType} ${colorBorder}`;
 
   return {
     [componentCls]: {
       ...resetComponent(token),
       backgroundColor: headerBg,
       border: borderBase,
       borderBottom: 0,
-      borderRadius: `${collapsePanelBorderRadius}px`,
+      borderRadius: `${unit(collapsePanelBorderRadius)}`,
```
Comment (MadCcc) [inline components/collapse/style/index.ts:78]:
```suggestion
      borderRadius: collapsePanelBorderRadius,
```

**Comment by MadCcc** [comment]
lint 看下

**Comment by RedJue** [comment]
> lint 看下

我看看

**Review by MadCcc — APPROVED** [review]

**Comment by MadCcc** [comment]
image
有一个 diff

File: components/collapse/style/index.ts [hunk 2]
```diff
@@ -188,9 +192,9 @@ export const genBaseStyle: GenerateStyle<CollapseToken> = (token) => {
             paddingInlineStart: padding,
 
             [`> ${componentCls}-expand-icon`]: {
-              height: fontSizeLG * lineHeight,
+              height: fontHeight,
```
Comment (MadCcc) [inline components/collapse/style/index.ts:195]:
```suggestion
              height: fontHeightLG,
```
这里正确应该是 fontSizeLG * lineHeightLG，看了一下文字行高也不对，需要改一下

Comment (RedJue) [reply components/collapse/style/index.ts:195]: 1

File: components/collapse/style/index.ts [hunk 3]
```diff
@@ -95,7 +100,7 @@ export const genBaseStyle: GenerateStyle<CollapseToken> = (token) => {
           alignItems: 'flex-start',
           padding: headerPadding,
           color: colorTextHeading,
-          lineHeight,
+          lineHeight: lineHeightLG,
```
Comment (MadCcc) [inline components/collapse/style/index.ts:103]: 这里不用改哦，应该是 large 的 Collapse 才需要改

Comment (MadCcc) [reply components/collapse/style/index.ts:103]: https://github.com/ant-design/ant-design/pull/45862/files#diff-4fadcc09942a875859791f11134d4aed49b1d7245a8627f9aa5db5c0a25bcdd0R189
这里应该同时改 lineHeight

Comment (RedJue) [reply components/collapse/style/index.ts:103]: ok

**Comment by MadCcc** [comment]
image
这样改

**Comment by RedJue** [comment]
> image 这样改

好 我再改下

**Pull request merged**